client/app/models/contact.coffee

```diff
@@ -148,6 +148,26 @@ module.exports = class Contact extends Backbone.Model
         _.compact([pf, fn, mn, gn, sf]).join ' '
 
 
+    # We override toJSON to ensure the exported object is the a true object.
+    # Allow a options.toVCF boolean to sanitize the output before.
+    toJSON: ({toVCF} = {}) ->
```

Comment:
I am not a huge fan of overriding the `toJSON`  method, especially if we modify its signature.

Moreover, in my opinion the `toVCF` could be as following without putting to much stuff on it:
```coffeescript
toVCF: (callback) ->
    data = @toJSON()
    data.datapoints = data.datapoints.toJSON()
    # Isolate in a try..catch block to go outside main thread and pop error
    try
        # We remove the BDay datapoint if it doesn't match ISO 8601
        unless moment(data.bday, 'YYYY-MM-DD', true).isValid()
            delete data['bday']
            throw new Error "Malformed bday for #{data.id}"
    catch err
        console.error err
    @picturetoa (picture) -> callback VCardParser.toVCF data, picture
```